Assets/Scripts/StandardSamples/UI/Common/SampleMenu.cs

```diff
@@ -0,0 +1,316 @@
+        /// launches. Most of them should be hidden.
+        /// </param>
+        /// <param name="requiresAuthentication">
+        /// Indicates whether the SampleMenu requires a user be authenticated in
+        /// order for it to be used. Most of them require authentication.
+        /// </param>
+        protected SampleMenu(bool startsHidden = true, bool requiresAuthentication = true)
+        {
+            StartsHidden = startsHidden;
+            RequiresAuthentication = requiresAuthentication;
+        }
+
+        /// <summary>
+        /// Function that handles change in authentication.
+        /// </summary>
+        /// <param name="authenticated">
+        /// True if the state has changed to authenticated, false otherwise.
+        /// </param>
+        private void OnAuthenticationChanged(bool authenticated)
+        {
+            if (authenticated || !RequiresAuthentication)
+            {
+                Show();
+            }
+            else
+            {
+                Hide();
+            }
+        }
+
+        /// <summary>
+        /// When overriding, call this base implementation first.
+        /// </summary>
+        protected virtual void Awake()
+        {
+            // When awake is called, subscribe to the authentication change
+            // event on AuthenticationListener.
+            AuthenticationListener.Instance.AuthenticationChanged += OnAuthenticationChanged;
+        }
+
+        /// <summary>
+        /// When overriding, call this base implementation first.
+        /// </summary>
+        protected virtual void OnDestroy()
+        {
+            // When OnDestroy is called, remove the subscription from the 
+            // AuthenticationChanged event.
+            AuthenticationListener.Instance.AuthenticationChanged -= OnAuthenticationChanged;
+        }
+
+        /// <summary>
+        /// When overriding, call this base implementation first.
+        /// </summary>
+        protected virtual void OnEnable()
+        {
+            // If the scene should start hidden
+            if (StartsHidden)
+            {
+                Hide();
+            }
+        }
+
+        /// <summary>
+        /// When overriding, call this base implementation first.
+        /// </summary>
+        protected virtual void Update()
+        {
+            // If nothing is selected, set the selected (focused) control to be
+            // UIFirstSelected (if that field has been set).
+            SetSelected();
+        }
+
+        /// <summary>
+        /// Shows the SampleMenu. If overriding, make sure to call this base
+        /// implementation first.
+        /// </summary>
+        public virtual void Show()
+        {
+            Log($"Show() started");
+
+            // Don't do anything if already showing.
+            if (!Hidden) return;
+
+            if (null == UIParent)
+            {
+                Log($"UIParent for is null for some reason.");
```

Comment:
Should this return here?

Reply:
Nope. The `UIParent` parameter is something that is optionally set by a user from within Unity Editor's inspector window. It's not guaranteed that the `UIFirstSelected` field member is set, so even if it _is_ null, the show method should still continue to set the `gameObject` to be active.